### PR TITLE
Whitelisted modal content

### LINKED ISSUE #3178: [UI] Implement Whitelisted content of Manage Whitelist modal
## Specification
Implement Whitelisted tab content in Manage Whitelist modal. Try to reuse existing logic 
* including no addresses state
* Implement search logic
* Implement hover state for removal user

### Design (optional, might not be applicable)
https://www.figma.com/file/Phby7kEYihhvGgQn69A314/Verified-Recipients?node-id=79%3A14613
https://www.figma.com/file/Phby7kEYihhvGgQn69A314/Verified-Recipients?node-id=79%3A10674 - no addresses state

### PR DESCRIPTION
**Pull request by alicjakujawa**
## Description

This pr is adding content for whitelisted tab 

**New stuff** ✨

* UserCheckbox component
* WhitelistedAddresses component
* NoWhitelistedUsers component

**Changes** 🏗

* Updated Checkbox component (new styles, callback method, tooltip updates)

**Deletions** ⚰️

* Deletions go here

Screenshot 2022-03-08 at 12 03 49
Screenshot 2022-03-08 at 12 03 39
Screenshot 2022-03-11 at 15 02 01

Resolves #3178

### CODE REVIEW

**Review by ArmandoGraterol — CHANGES_REQUESTED** [review]
Nice work! What about the no addresses state thought?

FireShot Capture 548 - Verified Recipients – Figma - www figma com

It was mentioned in the issue that it's also included.

I think the items need to have more space between them, as per the specs:
FireShot Capture 551 - Verified Recipients – Figma - www figma com

Right now:
FireShot Capture 550 - Colony - localhost

**Review by arrenv — CHANGES_REQUESTED** [review]
@alicjakujawa It is looking really good.

In addition to what Armando raised, I also noticed that the search filter does not reset after clearing the input. See gif below:

search-filter

File: src/modules/core/components/Fields/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -125,9 +131,15 @@ const Checkbox = ({
     [disabled, handleOnChange, inputId, isChecked, name],
   );
 
+  useEffect(() => {
+    if (getDefaultValue) {
+      getDefaultValue(isChecked);
+    }
+  }, []);
```
[the remark this answers is not in the document]
Comment (alicjakujawa) [reply src/modules/core/components/Fields/Checkbox/Checkbox.tsx:138]: nope, i dont want additional deps here, i want fire it only once at the beginning

**Review by danbr — CHANGES_REQUESTED** [review]
UI looking pretty. 
The tooltips were temperamental. I found it difficult to make them appear. Without them, it's quite confusing to know which action is about to happen.

manageWhitelist

Perhaps it is not covered in this PR (i'll add my comment, just in case), but the colour of the 'Confirm' button should be red.
Screenshot 2022-03-09 at 16 05 03

**Comment by alicjakujawa** [comment]
@danbr @ArmandoGraterol @arrenv Made updates. I will handle tooltip issue in separate PR to not block other tasks from this feature. Also currently we always display annotation field, later on when we implement logic for both tabs, we would need to also create logic when/how to display annotation (as it wont always display)

**Review by ArmandoGraterol — APPROVED** [review]
Looking good! I think all of the changes were addressed :+1:

**Review by arrenv — APPROVED** [review]
Looks good to me. Considering that the tooltip will be handled in another issue.

**Pull request merged**